doozerlib/cli/plashet.py

```diff
@@ -545,15 +550,85 @@ def from_tags(config, brew_tag, embargoed_brew_tag, embargoed_nvr, signing_advis
                 package_name = build['package_name']
                 released_package_nvre_obj[package_name] = parse_nvr(to_nvre(build))
 
-        if not runtime.assembly:
+        component_builds: Dict[str, Dict] = {}  # candidate rpms for plashet; keys are rpm component names, values are Brew build dicts
+        pinned_nvrs: Dict[str, str] = {}  # rpms pinned to the runtime assembly; keys are rpm component names, values are nvrs
+
+        if not assembly:
             # Assemblies are disabled. We just need the latest tagged builds in the brew tag
             latest_builds = koji_proxy.listTagged(tag, latest=True, inherit=inherit, event=event, type='rpm')
+            component_builds = {build["name"]: build for build in latest_builds}
         else:
             # Assemblies are enabled. We need all tagged builds in the brew tag then find the latest ones for the runtime assembly.
+            if runtime.assembly_basis_event:
+                logger.warning(f'Constraining rpm search to stream assembly due to assembly basis event {runtime.assembly_basis_event}')
+                assembly = 'stream'
             tagged_builds = koji_proxy.listTagged(tag, latest=False, inherit=inherit, event=event, type='rpm')
-            latest_builds = find_latest_builds(tagged_builds, runtime.assembly)
-
-        for build in latest_builds:
+            latest_builds = find_latest_builds(tagged_builds, assembly)
+            component_builds = {build["name"]: build for build in latest_builds}
+
+            el_version_match = re.search(r"rhel-(\d+)", tag)
+            el_version = int(el_version_match[1]) if el_version_match else 0
+            if el_version:  # Only honor pinned rpms if this tag is relevant to a RHEL version
+                # Honor pinned rpm nvrs specified in assembly.members.rpms
+                for component in runtime.releases_config.releases[runtime.assembly].assembly.members.rpms:
```

Comment:
This doesn't appear to take into account assemblies inheriting from other assemblies via `basis.assembly`. I'd suggest a new method in assembly.py like `assembly_member_config`.

Reply:
Actually, I can't think of any reason not to put `is` in metadata. https://github.com/openshift/ocp-build-data-validator/pull/35/commits/2c5af0845b8c21245adf86b4eb85fb8560456700

This should let you call:
```
meta_config = assembly_metadata_config(runtime.get_releases_config(), runtime.assembly, distgit_key, Model())
if meta_config['is']: ....
```

This will pick up any inherited value.

Reply:
Yeah, that sounds easier and we can reuse the logic in different places.